### DISCUSSION TITLE
v2.20

### DISCUSSION THREAD

**Discussion by Ralim** [Ralim/IronOS#1498] [General]
# High level changes:
* First "full" release for PinecilV2
* Loots of documentation updates
* Documentation is [now nicely readable as a site](https://ralim.github.io/IronOS/GettingStarted/)
* A fair collection of bugfixes for PinecilV2
* Cold Junction Calibration was reworked and now occurs _at next boot_ to make it easier to perform when the device is cold

## What's Changed (the details)
* Cleanup getButtonState(), fix button handling bugs by @nyanpasu64 in https://github.com/Ralim/IronOS/pull/1331
* Ticktype by @Ralim in https://github.com/Ralim/IronOS/pull/1332
* USB PD EPR Basics by @Ralim in https://github.com/Ralim/IronOS/pull/1333
* Allow preheating iron during boot logo (fixes #1334) by @aaronjamt in https://github.com/Ralim/IronOS/pull/1338
* Updates to documents by @River-b in https://github.com/Ralim/IronOS/pull/1346
* Update translation_ES.json by @Meldinov in https://github.com/Ralim/IronOS/pull/1343
* Update translation_UK.json by @eslng in https://github.com/Ralim/IronOS/pull/1352
* document updates by @River-b in https://github.com/Ralim/IronOS/pull/1348
* document updates by @River-b in https://github.com/Ralim/IronOS/pull/1355
* Fixed formatting for CS & NL / addition to Readme by @discip in https://github.com/Ralim/IronOS/pull/1357
* update Debug info by @River-b in https://github.com/Ralim/IronOS/pull/1358
* Update README.md by @discip in https://github.com/Ralim/IronOS/pull/1359
* solves artifacts issue if .gif is exactly 1024 B by @discip in https://github.com/Ralim/IronOS/pull/1362
* Animation non block by @Ralim in https://github.com/Ralim/IronOS/pull/1347
* Update DebugMenu.md by @River-b in https://github.com/Ralim/IronOS/pull/1367
* Update DebugMenu.md by @River-b in https://github.com/Ralim/IronOS/pull/1366
* Pinecil V2 by @Ralim in https://github.com/Ralim/IronOS/pull/1341
* typo fix by @discip in https://github.com/Ralim/IronOS/pull/1368
* Add SC7 accelerometer for MHP30 by @Ralim in https://github.com/Ralim/IronOS/pull/1377
* Clarify and add details to documents by @River-b in https://github.com/Ralim/IronOS/pull/1383
* reorder debug menu . . . by @discip in https://github.com/Ralim/IronOS/pull/1380
* Bump timeout for menu actions by @Ralim in https://github.com/Ralim/IronOS/pull/1384
* translation_update by @discip in https://github.com/Ralim/IronOS/pull/1391
* New string translation by @federicodilo in https://github.com/Ralim/IronOS/pull/1392
* Update translation_FR.json by @HanaO00 in https://github.com/Ralim/IronOS/pull/1399
* Updated documents, fixed English Translation by @River-b in https://github.com/Ralim/IronOS/pull/1400
* moved_reset_settings by @discip in https://github.com/Ralim/IronOS/pull/1401
* Dmitrygribenchuk by @dmitrygribenchuk in https://github.com/Ralim/IronOS/pull/1402
* Removed unused libraries by @dmitrygribenchuk in https://github.com/Ralim/IronOS/pull/1404
* Calibrate CJC at next boot by @discip in https://github.com/Ralim/IronOS/pull/1398
* Generate a mkdocs github page + Migrate documentation links to use it by @Ralim in https://github.com/Ralim/IronOS/pull/1414
* Fixing up TS80P Build by @Ralim in https://github.com/Ralim/IronOS/pull/1429
* Update translation_SK.json by @OndroNR in https://github.com/Ralim/IronOS/pull/1427
* Update PL translation (v2.19) by @KITT3000 in https://github.com/Ralim/IronOS/pull/1431
* Update translation_NL.json by @phulstaert in https://github.com/Ralim/IronOS/pull/1433
* update translation_BE.json by @dmitrygribenchuk in https://github.com/Ralim/IronOS/pull/1439
* update translation_UK.json by @dmitrygribenchuk in https://github.com/Ralim/IronOS/pull/1440
* language update (follow-up on #1398) by @discip in https://github.com/Ralim/IronOS/pull/1432
* update translation_UK.json translation_RU.json translation_BE.json translation_PL.json by @dmitrygribenchuk in https://github.com/Ralim/IronOS/pull/1444
* cleared up some wording along with some typo fixes by @discip in https://github.com/Ralim/IronOS/pull/1451
* Update translation_RO.json by @discip in https://github.com/Ralim/IronOS/pull/1443
* Update translation_IT.json by @discip in https://github.com/Ralim/IronOS/pull/1442
* Update translation_RU.json by @JugglerLKR in https://github.com/Ralim/IronOS/pull/1452
* Update translation_UK.json by @eslng in https://github.com/Ralim/IronOS/pull/1453
* solving issue #1423 by @discip in https://github.com/Ralim/IronOS/pull/1459
* Update translation_DA.json by @t-xdk in https://github.com/Ralim/IronOS/pull/1457
* Language update (NO => NB) by @discip in https://github.com/Ralim/IronOS/pull/1463
* Update Chinese and Japanese translations by @alvinhochun in https://github.com/Ralim/IronOS/pull/1445
* Only send on display change by @Ralim in https://github.com/Ralim/IronOS/pull/1464
* update translation_BE.json translation_PL.json by @dmitrygribenchuk in https://github.com/Ralim/IronOS/pull/1468
* reworked CJC calibration appearance by @discip in https://github.com/Ralim/IronOS/pull/1473
* Improve I2C for PinecilV2 by @Ralim in https://github.com/Ralim/IronOS/pull/1436
* update translation by @dmitrygribenchuk in https://github.com/Ralim/IronOS/pull/1480
* fixed CJC calibration condition by @discip in https://github.com/Ralim/IronOS/pull/1481
* Update translation_IT.json by @federicodilo in https://github.com/Ralim/IronOS/pull/1483
* Refactor gui files by @Ralim in https://github.com/Ralim/IronOS/pull/1478
* Update translation_UK.json by @eslng in https://github.com/Ralim/IronOS/pull/1486
* Cleaning up translations a small amount by @Ralim in https://github.com/Ralim/IronOS/pull/1487
* Drop the fonts key by @github-actions in https://github.com/Ralim/IronOS/pull/1490
* Revision on IT translation by @federicodilo in https://github.com/Ralim/IronOS/pull/1492
* removed some redundant lines by @discip in https://github.com/Ralim/IronOS/pull/1494

## New Contributors
* @nyanpasu64 made their first contribution in https://github.com/Ralim/IronOS/pull/1331
* @aaronjamt made their first contribution in https://github.com/Ralim/IronOS/pull/1338
* @Meldinov made their first contribution in https://github.com/Ralim/IronOS/pull/1343
* @eslng made their first contribution in https://github.com/Ralim/IronOS/pull/1352
* @phulstaert made their first contribution in https://github.com/Ralim/IronOS/pull/1433
* @t-xdk made their first contribution in https://github.com/Ralim/IronOS/pull/1457

**Full Changelog**: https://github.com/Ralim/IronOS/compare/v2.19...v2.20

## Note for PinecilV2

As the flashing tools for the BL706 are still a _little bit of a mess_; there is a community effort going on to get [blisp](https://github.com/pine64/blisp) finished and ready for use. It's mostly there (command line only). As its use is still developing its not fully documented in the current IronOS docs. The community chat is definitely around to help with issues using the flasher. There will more more improvement in this space as time permits for development.

<hr /><em>This discussion was created from the release <a href='https://github.com/Ralim/IronOS/releases/tag/v2.20'>v2.20</a>.</em>

**Comment by JugglerLKR** [comment]
I'm new to githb. What is the correct procedure to get notified via email when english translation updates, so I can update russian translation day or two BEFORE @Ralim new release?

**Reply by Ralim** [reply]
Hia; so historically there _hasnt_ been a great way.
But one of the things I'm doing for 2.21 is moving to weblate where you can subscribe :)

**Comment by dsagdfgsFD** [comment]
Hi. I just want to report a finding i had today. 
I just bought a TS100. Supposedly with a STM32 clone. (DFU v. 3.48). 
I could not flash on windows. Tried different PCs and different cables and adapters. 
It "crashed" and became a brick. Then i tried the original firmware with succes. 

Later i tried on my mac, and it did it on second try with IronOS. :)

**Reply by aaronjamt** [reply]
Did you use the same firmware on the Mac as on Windows?

**Reply by dsagdfgsFD** [reply]
Yes. The newest v2.20. Same language also. 
I also tried different languages in the proces.

**Reply by dsagdfgsFD** [reply]
Info on my mac: 
Macbook Pro 15" 2016. MacOS Monterey (v 12.6.2)

**Comment by ZvEroBoY36** [comment]
Hi Ralim. I became a recent owner of TS101, please tell me if there will be another firmware version for this soldering iron in the future? I have already downloaded the latest version for my soldering iron, but I want some more improvements. Thank you.

**Reply by River-Mochi** [reply]
@ZvEroBoY36 
there was a ticket opened asking for porting to miniware ts101
see Ralim response in ticket here:
  https://github.com/Ralim/IronOS/issues/1420#issuecomment-1334529802

you can scroll up  to see all of Ralim's responses while he looked at the device.
